### PR TITLE
BED-4612: : Citrix RDP Configuration UI

### PR DESCRIPTION
**Pull request by specter-flq**
## Description

This PR adds a new BloodHound Configuration panel under the Administration menu, it will allow to toggle Citrix RDP support. This menu option/route will only work for Administrators and Power Users.

## Motivation and Context

We want to create a new configuration page that allows users to adjust runtime parameters available via the /api/v2/config/ endpoint.

## How Has This Been Tested?

Added test coverage, plus lots of manual testing against mocked endpoints with MSW

## Screenshots (optional):

Screenshot 2024-09-13 at 2 11 26 PM

Screenshot 2024-09-13 at 2 11 38 PM

https://github.com/user-attachments/assets/2202ed6e-e05a-46be-9dc2-fb9429e05fc2

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    const [isOpenDialog, setIsOpenDialog] = useState(false);
+
+    const { addNotification } = useNotifications();
+    const { data, isFetching } = useGetConfiguration();
+    const updateConfiguration = useUpdateConfiguration();
+
+    const citrixRDPconfigurationEnabled = parseCitrixConfiguration(data)?.value.enabled;
+
+    // optimistically update switch state during pending mutations
+    const haveUnsettledRequests = updateConfiguration.isLoading || isFetching;
```
Comment (brandonshearin) [inline packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfiguration.tsx:39]: wicked

**Review by brandonshearin — APPROVED** [review]

File: packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfirmDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+const CitrixRDPConfirmDialog: FC<CitrixRDPConfirmDialogProps> = ({ open, futureSwitchState, onCancel, onConfirm }) => {
+    const theme = useTheme();
+
+    return (
+        <Dialog
+            open={open}
+            maxWidth='sm'
+            aria-labelledby='citrix-rdp-alert-dialog-title'
+            aria-describedby='citrix-rdp-alert-dialog-description'>
```
Comment (urangel) [inline packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfirmDialog.tsx:39]: Thank you for adding aria attributes! Do we need to add the `id` attribute to the element that holds the description for `'citrix-rdp-alert-dialog-description'`?

File: packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfirmDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                {dialogTitle}
+            </DialogTitle>
+            <DialogContent sx={{ paddingBottom: 0 }}>
+                <Typography sx={{ paddingBottom: '16px', whiteSpace: 'break-spaces', fontSize: '12px' }}>
+                    {futureSwitchState ? enabledDialogDescription : disabledDialogDescription}
+                </Typography>
+                <Typography sx={{ fontSize: '12px' }}>
+                    Select <b>`Confirm`</b> to proceed. Changes will be reflected upon completion of next analysis.
+                </Typography>
+                <Typography sx={{ fontSize: '12px' }}>
```
Comment (urangel) [inline packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfirmDialog.tsx:50]: The decimal values may look wonky but I would recommend converting these `px` unit values to unit values based off of `rem`. For example, 1rem for 16px, 1.25rem for 20px, and 0.75 rem for 12px.

File: packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfirmDialog.tsx (hunk 2 above)
Comment (specter-flq) [reply packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfirmDialog.tsx:39]: good catch thank you, updated.

File: cmd/ui/src/views/BloodHoundConfiguration/BloodHoundConfiguration.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Box, Link, Typography } from '@mui/material';
+import { PageWithTitle, CitrixRDPConfiguration } from 'bh-shared-ui';
+
+const BloodHoundConfiguration = () => {
+    return (
+        <PageWithTitle
+            title='Bloodhound Configuration'
```
[the remark this answers is not in the document]
Comment (specter-flq) [reply cmd/ui/src/views/BloodHoundConfiguration/BloodHoundConfiguration.tsx]: good eye, updated.

File: packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfiguration.tsx (hunk 1 above)
Comment (specter-flq) [reply packages/javascript/bh-shared-ui/src/components/CitrixRDPConfiguration/CitrixRDPConfiguration.tsx:39]: that was all Wes!

**Review by urangel — APPROVED** [review]
Great work! 🚀

**Pull request merged**